overlay/tensorflow_validation/validation.py

```diff
@@ -86,22 +86,27 @@ def validate_gis_joins_multithreaded(self, gis_joins: list) -> list:
     def validate_gis_join(self, gis_join: str, querier: Querier, model: tf.keras.Model, is_concurrent: bool) -> float:
         # Query MongoDB for documents matching GISJOIN
 
-        if self.validation_budget.budget_type == BudgetType.STATIC_BUDGET:
-            budget: StaticBudget = self.validation_budget.static_budget
+        if self.request.validation_budget.budget_type == BudgetType.STATIC_BUDGET:
+            budget: StaticBudget = self.request.validation_budget.static_budget
             limit: int = budget.strata_limit
             sample_rate: float = budget.sample_rate
 
             documents = querier.spatial_query(
-                self.collection,
+                self.request.collection,
                 gis_join,
-                self.feature_fields,
-                self.label_field,
+                self.request.feature_fields,
+                self.request.label_field,
                 limit,
                 sample_rate
             )
 
             # Load MongoDB Documents into Pandas DataFrame
             features_df = pd.DataFrame(list(documents))
+
```

Comment:
Once we have the Documents in-memory as a Pandas DataFrame, we need to close the connection ASAP so the router doesn't get overwhelmed with open connections. I've noticed this happening and the router crashing if we wait to close the connection until after we evaluate the model.